components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -307,13 +311,6 @@ class GeckoEngineSession(
             session: GeckoSession,
             request: NavigationDelegate.LoadRequest
         ): GeckoResult<AllowOrDeny> {
-            // TODO use onNewSession and create window request:
-            // https://github.com/mozilla-mobile/android-components/issues/1503
-            if (request.target == GeckoSession.NavigationDelegate.TARGET_WINDOW_NEW) {
-                geckoSession.loadUri(request.uri)
-                return GeckoResult.fromValue(AllowOrDeny.DENY)
-            }
```

Comment:
I wonder if we should return ALLOW here instead of removing the block. We didn't execute the following block for `TARGET_WINDOW_NEW` before and I wonder if it makes sense to do it now - especially since the following code (like interceptors) do not know about the target? 🤔

Reply:
Yeah, good call. Updated and re-tested.